lib/rules/no-use-before-define.js

```diff
@@ -136,6 +136,8 @@ function isInInitializer(variable, reference) {
 
 module.exports = {
     meta: {
+        type: "suggestion",
```

Comment:
If this rule only checked TDZ issues, I’d advocate for it to be a problem. Because it also checks hoisting, I could go either way. The TDZ messages are in the “fix this, your program is broken” category, and those lose significance because the hosting messages are appropriately considered suggestions.

Reply:
Hmm, yeah, this is a bit sticky. I think we should probably go with the most severe warning that comes out of the rule, so that would make this a problem.